pkg/mtu/manager.go

```diff
@@ -4,19 +4,30 @@
 package mtu
 
 import (
+	"bytes"
 	"context"
 	"fmt"
+	"net"
 	"slices"
+	"sync/atomic"
 
 	"github.com/cilium/cilium/pkg/datapath/tables"
 	"github.com/cilium/cilium/pkg/defaults"
+	ipamOption "github.com/cilium/cilium/pkg/ipam/option"
+	v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/node/addressing"
 
 	"github.com/cilium/hive/cell"
 )
 
 type MTUManager struct {
 	mtuParams
 	Config *Configuration
+
+	localNode     atomic.Pointer[v2.CiliumNode]
+	localNodeInit chan struct{}
```

Comment:
What about returning the updated localNode from the observer job directly into the channel?

Doing that allows to have a single `localNode chan *v2.CiliumNode` instead of an `atomic.Pointer` plus a channel to notify an update. The `consideredDevices` method can be changed to accept a `*v2.CiliumNode` parameter (at the first iteration it will be nil, but that's not a problem, as it is already properly handled).

Reply:
I am not sure that makes it easier. I essentially need a local node store but for `*v2.CiliumNode ` I always want the latest known version of the CiliumNode for the local node. If I were to use the channel I get all of the intermediate states which in this case don't matter.